scripts/repoHandlers.js

```diff
@@ -72,40 +72,57 @@ function updateTrackedReposInBitbucketOrg(trackedRepos, websiteUrl) {
  * @param {boolean} isOrg - Whether github organization or user.
  */
 function updateTrackedReposInGitHub(trackedRepos, websiteUrl, ownerType) {
-	const allRepo = ownerType == 'org' ? document.getElementById('org-repositories') : document.getElementById('user-repositories-list');
-	const repoUrl = Array.from(allRepo.querySelectorAll('a[itemprop="name codeRepository"]'));
+    let repoList;
+    let newUI = false;
 
+    // Check if selectors for new UI exist
+    if (document.querySelector('[data-testid="list-view-items"]')) {
+        repoList = ownerType === 'org' ? document.querySelectorAll('[data-testid="list-view-items"] > li') : document.querySelectorAll('[data-filterable-for="your-repos-filter"] > li');
+		newUI = true;
+    } else {
+        const allRepo = ownerType === 'org' ? document.getElementById('org-repositories') : document.getElementById('user-repositories-list');
+        repoList = Array.from(allRepo.querySelectorAll('a[itemprop="name codeRepository"]'));
+    }
+    repoList.forEach(repoItem => {
+		let repoUrl;
+		let repoLink
+		if (newUI){
+			repoLink = ownerType == 'org' ? repoItem.querySelector('[data-testid="listitem-title-link"]') : repoItem.querySelector('a[itemprop="name codeRepository"]');
+			if (!repoLink) return;
+			repoUrl = repoLink.getAttribute('href');
+		} else {
+			repoUrl = repoItem.getAttribute('href')
+		}
+		const repoName = repoUrl.split('/').pop();
+		
+        if (trackedRepos.includes(repoName)) {
+            const checkElement = repoItem.querySelector('.trackLogo');
+            if (checkElement) {
+                checkElement.remove();
+            }
 
-	repoUrl.forEach((item) => {
-		const link = item.getAttribute('href').split('/');
-		const repoName = link[link.length - 1];
+            const img = document.createElement("img");
+            img.classList.add('trackLogo');
+            img.src = `${websiteUrl}/favicon.ico`;
+            img.style.width = '15px';
+            img.style.height = '15px';
 
-		if (trackedRepos.includes(repoName)) {
-			const checkElement = item.getElementsByClassName('trackLogo')[0];
-			if (checkElement) {
-				// TODO: Ideally, we should only need to add the element when there is none present
-				checkElement.remove();
-			}
-			const img = document.createElement("img");
-			img.setAttribute('class', 'trackLogo');
-			const beforePsuedoElement = document.createElement('a');
-			img.src = `${websiteUrl}/favicon.ico`;
-			img.style.width = '15px'
-			img.style.height = '15px'
+            const linkElement = document.createElement('a');
```

Comment:
`linkElement` is not a good name. Try `vibinexLogoElement` or `insertBeforeElement` - they do a better job at explaining what the element really is. (The `<a>` type of the element is not its defining characteristic).